src/tsolvers/arraysolver/ArraySolver.cc

```diff
@@ -450,10 +450,10 @@ std::vector<ArraySolver::LemmaConditions> ArraySolver::collectLemmaConditions(Lo
         // TODO: Figure out better way how to compute all candidates for lemmas
         for (auto first : selects) {
             ERef firstRoot = getRoot(first);
+            NodeRef arrayFirst = getNodeRef(getRoot(getArrayFromSelect(first)));
             for (auto secondIt = selects.begin(); *secondIt != first; ++secondIt) {
                 ERef second = *secondIt;
                 if (firstRoot == getRoot(second)) { continue; } // The selects are already the same, no lemma needed
-                NodeRef arrayFirst = getNodeRef(getRoot(getArrayFromSelect(first)));
                 NodeRef arraySecond = getNodeRef(getRoot(getArrayFromSelect(second)));
                 if (arrayFirst == arraySecond or getIndexedRepresentative(arrayFirst, index) == getIndexedRepresentative(arraySecond, index)) {
                     std::unordered_set<PTRef, PTRefHash> undecidedEqualities;
```

Comment:
What happens if this hash is made to reuse its storage? I.e. move it outside the loops and just call `undecidedEqualities.clear()` here?

Reply:
We cannot do that. The map is currently being moved into `LemmaConditions`.

Reply:
Yes, but after moving and clearing it is again in a valid state. It may potentially reuse previously dynamically allocated storage, it depends on implementation.